docs/adr-dependencies.md

```diff
@@ -0,0 +1,97 @@
+# usage of Dependencies, peerDependencies, devDependencies in package.json
+
+## Context
+
+Dependencies are specified using the package.json file in library repositories.
+We have more and more libraries written and used by projects.
+
+We use different distribution formats for our libraries:
+
+- UMD for some (react-components, react-cmf, etc ...)
+- mix of transpiled JS and SASS files which need shared config to make it work for others
+
+## Problems
+
+We see commits with a mix of libraries being added to `peerDependencies` and `dependencies` without real guidelines.
+Each change in peerDependencies leads to [a major release](https://github.com/semver/semver/issues/502).
+
+We do not know when to put a dependency in the `"peerDependencies"` or just in the `"dependencies"` attribute of the package.json file.
+
+With yarn, `peerDependencies` are hard to manage. Yarn only outputs a warning the first time it encounters the dependency, when it needs to really install it in your node_modules. It's easy to miss and to forget. We have also seen some false positive warnings.
+
+With npm, peerDependency management depends on the version used:
+
+- npm 1,2,7,8 install peerDependencies
+- npm 3,4,5,6 do not install peerDependencies
+
+The documentation of npm only covers the behavior of npm, not the definition and actual usage of these peer dependencies.
+
+We know our project can't use npm > 7.0 because we have some issues in our peerDependencies definitions! 
+npm > 7 requires all peer dependencies to be at the same level, the same version. That needs to be fixed on our end.
+
+## Solutions
+
+Define a clear guideline for dependency requirements in the package.json files of libraries.
+
+### Guideline
+
+**`"devDependencies"`**
+
+Add **a-dependency** under `devDependencies` if **a-dependency** is used only to build, debug, or because A is also a peerDependencies.
+
+Basically use `devDependencies` if this dependency has no impact on runtime.
+
+examples:
+
+- @talend/scripts-core
+- @talend/scripts-preset-react-lib
+- i18next-scanner
+- cross-env
+- react (if in peerDependencies)
+
+Type dependencies (usually in the format `@types/some-js-library`) can be added to `devDependencies` only if the exported types for the library you're working on do not depend on it.
+
+```javascript
+import { LibType } from 'some-js-library';
+
+// If this export is available in the bundle, then your bundle has an actual dependency to LibType
+export myLibType = LibType & { isActive: boolean };
+```
```

Comment:
```suggestion
```typescript
import { LibType } from 'some-js-library';

// If this export is available in the bundle, then your bundle has an actual dependency to LibType
export type myLibType = LibType & { isActive: boolean };
```
```